step-plans/step-plans-core/src/main/java/step/core/execution/ExecutionEngineRunner.java

```diff
@@ -233,13 +234,13 @@ private void resolveInnerPlanAndFunctionArtefacts(Plan plan) {
 	private ReportNode execute(Plan plan, ReportNode rootReportNode) throws ProvisioningException, DeprovisioningException {
 		ArtefactHandlerManager artefactHandlerManager = executionContext.getArtefactHandlerManager();
 		AbstractArtefact root = plan.getRoot();
-		artefactHandlerManager.createReportSkeleton(root, rootReportNode);
+		artefactHandlerManager.createReportSkeleton(root, rootReportNode, ParentSource.MAIN);
 
 		// Provision the resources required for the execution before starting the execution phase
 		provisionRequiredResources();
 		try {
 			updateStatus(ExecutionStatus.RUNNING);
-			return artefactHandlerManager.execute(root, rootReportNode);
+			return artefactHandlerManager.execute(root, rootReportNode, ParentSource.MAIN);

```

Comment:
Does this mean that we don't support other sources for the root element?

Reply:
no it doesn't. the "root" artefact may have before,after.. The parent source of the root artefact is the plan itself which can only have one and MAIN root artefact.